akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -150,36 +180,54 @@ final class AskableActorRef(val actorRef: ActorRef) extends AnyVal {
    * INTERNAL API: for binary compatibility
    */
   protected def ask(message: Any, timeout: Timeout): Future[Any] =
-    internalAsk(message, timeout, ActorRef.noSender)
+    internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
 
   def ask(message: Any)(implicit timeout: Timeout, sender: ActorRef = Actor.noSender): Future[Any] =
+    internalAsk(_ ⇒ message, timeout, sender)
+
+  def ask(message: ActorRef ⇒ Any)(implicit timeout: Timeout): Future[Any] =
+    internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
+
+  def ask(message: ActorRef ⇒ Any)(implicit timeout: Timeout, sender: ActorRef): Future[Any] =
     internalAsk(message, timeout, sender)
 
   /**
    * INTERNAL API: for binary compatibility
    */
   protected def ?(message: Any)(implicit timeout: Timeout): Future[Any] =
-    internalAsk(message, timeout, ActorRef.noSender)
+    internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
 
   def ?(message: Any)(implicit timeout: Timeout, sender: ActorRef = Actor.noSender): Future[Any] =
+    internalAsk(_ ⇒ message, timeout, sender)
+
+  def ?(message: ActorRef ⇒ Any)(implicit timeout: Timeout): Future[Any] =
+    internalAsk(message, timeout, ActorRef.noSender)
+
+  def ?(message: ActorRef ⇒ Any)(implicit timeout: Timeout, sender: ActorRef): Future[Any] =
     internalAsk(message, timeout, sender)
```

Comment:
Other options:
- provide a separate import for this use-case as a `richAsk` or `extendedAsk`;
- choose a different name:
  - `??` -- too close to `???`;
  - `>?<` -- takes longer to write;
  - `=?>` -- welcome to "this is why nobody likes symbolic operators" zone.

This also raises a question of consistency of usage across `akka-actor` and `akka-typed`, considering that the latter already provides a `?(Any => ActorRef)(Timeout)` option.

cc @rkuhn: any thoughts?